tiledb/sm/query/writers/global_order_writer.cc

```diff
@@ -598,27 +763,42 @@ Status GlobalOrderWriter::global_write() {
     return Status::Ok();
   }
 
-  // Set new number of tiles in the fragment metadata
-  auto new_num_tiles = frag_meta->tile_index_base() + tile_num;
-  throw_if_not_ok(frag_meta->set_num_tiles(new_num_tiles));
-
   // Compute coordinate metadata (if coordinates are present)
-  RETURN_CANCEL_OR_ERROR_ELSE(
-      compute_coords_metadata(tiles, frag_meta), clean_up(uri));
+  auto mbrs = compute_mbrs(tiles);
 
   // Compute tile metadata.
-  RETURN_CANCEL_OR_ERROR_ELSE(
-      compute_tiles_metadata(tile_num, tiles), clean_up(uri));
+  RETURN_CANCEL_OR_ERROR(compute_tiles_metadata(tile_num, tiles));
 
   // Filter all tiles
-  RETURN_CANCEL_OR_ERROR_ELSE(filter_tiles(&tiles), clean_up(uri));
+  RETURN_CANCEL_OR_ERROR(filter_tiles(&tiles));
 
-  // Write tiles for all attributes
-  RETURN_CANCEL_OR_ERROR_ELSE(
-      write_all_tiles(frag_meta, &tiles), clean_up(uri));
+  for (uint64_t idx = 0; idx < tile_num;) {
+    auto frag_meta = global_write_state_->frag_meta_;
+    auto num = num_tiles_to_write(idx, tile_num, tiles);
 
-  // Increment the tile index base for the next global order write.
-  frag_meta->set_tile_index_base(new_num_tiles);
+    // Set new number of tiles in the fragment metadata
+    auto new_num_tiles = frag_meta->tile_index_base() + num;
+    throw_if_not_ok(frag_meta->set_num_tiles(new_num_tiles));
+
+    if (new_num_tiles == 0) {
+      throw GlobalOrderWriterStatusException(
+          "Fragment size doesn't allow to write a single tile");
```

Comment:
Can we be more specific with this error? Maybe something like:
```suggestion
          "Fragment size is too small to write a single tile");
```